### PR TITLE
RDISCROWD-5242: Enable task updates for regular coowner

### PR DESCRIPTION
**Pull request by dchhabda**
*Issue number of the reported bug or feature request: #<number>*

**Describe your changes**
A clear and concise description of the changes you have made.

**Testing performed**
Describe the testing you have performed to ensure that the bug has been addressed, or that the new feature works as planned.

**Additional context**
Add any other context about your contribution here.

### CODE REVIEW

File: pybossa/auth/task.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def _read(self, user, task=None):
         return user.is_authenticated
 
     def _update(self, user, task):
-        return self._only_admin_or_subadminowners(user, task)
+        return self._only_admin_or_owners(user, task)
```
Comment (XiChenn) [inline pybossa/auth/task.py:44]: changing the _update implementation will affect all code calling ```ensure_authorized_to('update', project)```. Probably this is not what we want.

File: pybossa/auth/task.py [hunk 2]
```diff
@@ -65,6 +65,20 @@ def _only_admin_or_subadminowners(self, user, task=None, project_owners_ids=None
             project_owners_ids = project.owners_ids
         return user.admin or (user.subadmin and user.id in project_owners_ids)
 
+    def _only_admin_or_owners(self, user, task=None, project_owners_ids=None):
```
Comment (XiChenn) [inline pybossa/auth/task.py:68]: If ```def _update(self, user, task):``` doesn't need to change then this function is not needed.

**Review by XiChenn — CHANGES_REQUESTED** [review]

File: pybossa/auth/task.py (hunk 1 above)
Comment (dchhabda) [reply pybossa/auth/task.py:44]: This is under task auth, hence would be applied to task and won't affect all code.  call to `ensure_authorized_to('update', project)` would route to project auth [here](https://github.com/bloomberg/pybossa/blob/main/pybossa/auth/project.py#L61)

File: pybossa/auth/task.py (hunk 2 above)
Comment (dchhabda) [reply pybossa/auth/task.py:68]: As mentioned above, I think we need _update, hence this function also reqd.

**Pull request merged**